bundles/statistics/statsgrid2016/components/Legend.js

```diff
@@ -120,8 +123,17 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.Legend', function(sandbox, loca
             }
             //legend
             legendContainer.html( legendUI );
+        }); //_createLegend
+    },
 
-            var edit = container.find('.edit-legend');
+    /****** PRIVATE METHODS ******/
+    /**
+     * Adds functionality to edit classification button.
+     */
+    _createEditClassificationListener : function() {
+        var edit = this._element.find('.edit-legend');
+        if (edit) {
```

Comment:
This is always true since find always returns an object. To check if something was actually hit you need to check the length property.